### PR TITLE
feat(gui-v2): add smtp rejectUnauthorized

### PR DESCRIPTION
**Pull request by wingkwong**
## Change Summary

ref: #2276

- add `rejectUnauthorized` option for smtp
- change secure to checkbox

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

image

### CODE REVIEW

**Review by pranavxc — CHANGES_REQUESTED** [review]

File: packages/nocodb/src/lib/plugins/smtp/SMTP.ts [hunk 1]
```diff
@@ -14,21 +14,19 @@ export default class SMTP implements IEmailAdapter {
...
-      // from: this.input.from,
-      // options: {
       host: this.input?.host,
       port: parseInt(this.input?.port, 10),
-      secure: this.input?.secure === 'true',
-      ignoreTLS:
-        typeof this.input?.ignoreTLS === 'boolean'
-          ? this.input?.ignoreTLS
-          : this.input?.ignoreTLS === 'true',
+      secure: this.input?.secure,
```
[the remark this answers is not in the document]
Comment (wingkwong) [reply packages/nocodb/src/lib/plugins/smtp/SMTP.ts:19]: Covered here 

https://github.com/nocodb/nocodb/blob/b13297030683963b0a0eb94a422f489d004d5f10/packages/nc-gui-v2/components/dashboard/settings/app-store/AppInstall.vue#L106-L111

[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/lib/plugins/smtp/SMTP.ts:19]: What about the backend while initializing SMTP? For example if value is `'false'`

[the remark this answers is not in the document]
Comment (wingkwong) [reply packages/nocodb/src/lib/plugins/smtp/SMTP.ts]: That's a good point. Added it back.

**Review by pranavxc — APPROVED** [review]

**Comment by wingkwong** [comment]
Skipping v1 cypress.

**Pull request merged**